### PR TITLE
fix: judge file format by server's content-type header

### PR DESCRIPTION
**Pull request by qiu8310**

### CODE REVIEW

**Comment by qiu8310** [comment]
related to #106

**Comment by qiu8310** [comment]
There is another problem, can't figure out why:

The first `runTest` is success, but second failed.

image

**Comment by qiu8310** [comment]
It looks like the error is caused by the warning: 

`WARNING: IPC handle "/Users/Mora/Workspace/mora/example-ts-vscode-extension/vscode-test/sample/.vscode-test/user-data/1.63.2-main.sock" is longer than 103 chars, try a shorter --user-data-dir`

switch the project to a shorter folder, all tests is passed.

File: lib/download.ts [hunk 1]
```diff
@@ -86,6 +86,9 @@ async function downloadVSCodeArchive(options: DownloadOptions) {
 
 	const download = await request.getStream(url);
 	const totalBytes = Number(download.headers['content-length']);
+	const contentType = download.headers['content-type'];
+	const isZip = contentType ? contentType === 'application/zip' : url.endsWith('.zip');
```
Comment (connor4312) [inline lib/download.ts:90]:
```suggestion
	const isZip = download.headers['content-type'] === 'application/zip' ?? url.endsWith('.zip');
```

**Review by connor4312 — APPROVED** [review]
Thanks for looking into this! Not sure why it didn't cause issues on my machine. One small cleanup comment

File: lib/download.ts (hunk 1 above)
Comment (qiu8310) [reply lib/download.ts]: `download.headers['content-type'] === 'application/zip'` is always `true` or `false`, so `url.endsWith('.zip')` will never be reached.

**Pull request merged**